### PR TITLE
adding support to custom pricing configuration in helm chart

### LINKED ISSUE #108: Custom pricing configmap
# Description
OpenCost supports custom pricing for on-premise environments [here](https://www.opencost.io/docs/configuration/on-prem#custom-pricing-using-the-opencost-helm-chart). You can introduce this pricing by deploying a configmap in the environment sadly this isn't integrated into the helm chart and is deployed separately. If you go about it the way it is now it's not explicitly clear in the values as you have to introduce them as extraEnv and ExtraMount
```opencost:
  exporter:
    extraEnv:
      # The default config path is read only, for customizing we have to swap spots.
      CONFIG_PATH: "/tmp/custom-config"
    extraVolumeMounts:
      - mountPath: /tmp/custom-config
        name: custom-configs
extraVolumes:
  - name: custom-configs
    configMap:
      name: opencost-conf
```

# Expected Behavior
As an improvement, we should include the configmap for users to be defined in their values as they see fit with the possibility to use their own configmap if they want to.

### PR DESCRIPTION
**Pull request by HoussemCharf**
Closes #108

### CODE REVIEW

**Comment by mattray** [comment]
This looks like it duplicates the efforts of https://github.com/opencost/opencost-helm-chart/pull/107. Could you check that patch and see if there's anything in it this one needs? I do like this patch's use use of `opencost.onpremisemode.enabled`, so I'm leaning more towards it after it gets cleaned up.

**Comment by tamalsaha** [comment]
`opencost.onpremisemode.enabled` is semantically wrong for what I was doing in #107 . 

The metricsConfigs.disabledMetrics is necessary to avoid duplicate KMS v2 metrics.

**Comment by tamalsaha** [comment]
@HoussemCharf, yes. My pr tackles unrelated issues. Those are not related to onpremises deployment.

**Review by toscott — CHANGES_REQUESTED** [review]
I love the idea of simplifying the barrier for using custom pricing. Thank you for the suggestion and taking a stab at it!

I've made a few comments. Once addressed we can get this merged.

File: charts/opencost/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{{- if and .Values.opencost.onpremisemode.use_values_map .Values.opencost.onpremisemode.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Values.opencost.onpremisemode.configmap_name }}
+data:
+  default.json: |-
```
[the remark this answers is not in the document]
Comment (HoussemCharf) [reply charts/opencost/templates/configmap.yaml:7]: Good catch

File: charts/opencost/values.yaml [hunk 2]
```diff
@@ -129,6 +129,28 @@ opencost:
     # -- Any extra environment variables you would like to pass on to the pod
     extraEnv: {}
       # FOO: BAR
+  onpremisemode:
```
[the remark this answers is not in the document]
Comment (HoussemCharf) [reply charts/opencost/values.yaml:132]: sounds good

File: charts/opencost/values.yaml [hunk 3]
```diff
@@ -129,6 +129,28 @@ opencost:
...
+    # -- Enables custom pricing for on-premise setup.
+    enabled: false
+    configmap_name: custom-pricing-model
+    # -- Path for the pricing configuration.
+    config_path: /tmp/custom-config
+    # -- Configures the pricing model provided in the values file.
+    use_values_map: true
```
[the remark this answers is not in the document]
Comment (HoussemCharf) [reply charts/opencost/values.yaml:139]: sounds good

**Comment by toscott** [comment]
Awesome!
I plan to review, retest and get merged tonight

File: charts/opencost/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{{- if and .Values.opencost.customPricing.createConfigmap .Values.opencost.customPricing.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Values.opencost.customPricing.configmapName }}
+data:
+  default.json: |-
+    {
```
Comment (toscott) [inline charts/opencost/templates/configmap.yaml:8]: After the changes, this is still not formatting correctly. The trailing comma is causing parsing issues. 

> ERR error accessing cloud provider configuration: models.CustomPricing.ReadString: expects " or n, but found }, error found in #10 byte of ...| "0.01",

I think we can convert the block directly to json, then quote it.
`  default.json: {{ .Values.opencost.customPricing.costModel | toJson | quote }}`

Taking this approach will require explicit quoting on all of the numeric values in the values.yaml file though. There might be a more elegant approach of course.

File: charts/opencost/templates/configmap.yaml
Comment (toscott) [inline charts/opencost/templates/configmap.yaml:1]: While reviewing some of the other pull requests, its clear we are going to end up with multiple configmaps. Planning for the future, can you rename this file to configmap-custom-pricing.yaml? 

Thanks!

**Review by toscott — CHANGES_REQUESTED** [review]

File: charts/opencost/templates/configmap.yaml (hunk 4 above)
Comment (HoussemCharf) [reply charts/opencost/templates/configmap.yaml:8]: I kept reaching the same error with the method mentioned above

**Review by toscott — APPROVED** [review]
The last round of changes appears to be working well! Thanks for the contribution @HoussemCharf. Waiting on the CI flows to finish and I'll get this merged.

**Pull request merged**